src/elements/ngm-nav-tools.ts

```diff
@@ -67,12 +94,95 @@ export class NgmNavTools extends LitElementI18n {
   }
 
   flyToHome() {
-    if (!this.scene) return;
-    this.scene.camera.flyTo(DEFAULT_VIEW);
+    if (!this.viewer) return;
+    this.viewer.camera.flyTo(DEFAULT_VIEW);
+  }
+
+  toggleReference() {
+    if (!this.eventHandler) return;
+    if (this.showRefPoint) {
+      this.eventHandler.removeInputAction(ScreenSpaceEventType.MOUSE_MOVE);
+      this.eventHandler.removeInputAction(ScreenSpaceEventType.LEFT_DOWN);
+      this.eventHandler.removeInputAction(ScreenSpaceEventType.LEFT_UP);
+      this.removeReference();
+    } else {
+      this.eventHandler.setInputAction(debounce(event => this.onMoveRef(event), 250), ScreenSpaceEventType.MOUSE_MOVE);
+      this.eventHandler.setInputAction(event => this.onLeftDown(event), ScreenSpaceEventType.LEFT_DOWN);
+      this.eventHandler.setInputAction(() => this.onLeftUp(), ScreenSpaceEventType.LEFT_UP);
+      this.setReference();
+    }
+  }
+
+  setReference(position?: Cartesian3) {
+    const center = position || pickCenterOnMapOrObject(this.viewer!.scene);
+    if (!center) {
+      showWarning(i18next.t('nav_tools_out_glob_warn'));
+      return;
+    }
+    this.showRefPoint = true;
+    this.refIcon.position = <any>center;
+    const cam = this.viewer!.camera;
+    this.refIcon.show = true;
+    this.refIcon.viewFrom = <any>eastNorthUp(center, cam.position);
+    if (!position) {
+      const transform = Transforms.eastNorthUpToFixedFrame(center);
+      cam.lookAtTransform(transform);
+    }
+    document.addEventListener('keydown', this.ctrlListener);
+
+  }
+
+  removeReference() {
+    document.removeEventListener('keydown', this.ctrlListener);
+    this.showRefPoint = false;
+    this.refIcon.show = false;
+    this.viewer!.scene.camera.lookAtTransform(Matrix4.IDENTITY);
+    this.viewer!.trackedEntity = undefined;
+  }
+
+  ctrlListener = (evt) => {
+    if (evt.key !== 'Control') return;
+    this.removeReference();
+    showWarning(i18next.t('nav_tools_ctrl_disable_warn'));
+  }
+
+  onLeftDown(event) {
+    const pickedObject = this.viewer!.scene.pick(event.position);
+    if (pickedObject && pickedObject.id && pickedObject.id.id === this.refIcon.id) {
+      this.viewer!.scene.screenSpaceCameraController.enableInputs = false;
+      this.eventHandler!.setInputAction(event => this.onMoveRef(event), ScreenSpaceEventType.MOUSE_MOVE);
+      this.viewer!.scene.camera.lookAtTransform(Matrix4.IDENTITY);
+      this.viewer!.trackedEntity = undefined;
+      this.moveRef = true;
+    }
+  }
+
+  onLeftUp() {
+    if (!this.moveRef) return;
+    this.moveRef = false;
+    this.viewer!.trackedEntity = this.refIcon;
+    this.viewer!.scene.screenSpaceCameraController.enableInputs = true;
+    // for better performance
+    this.eventHandler!.setInputAction(debounce(event => this.onMoveRef(event), 250), ScreenSpaceEventType.MOUSE_MOVE);
+  }
+
+  onMoveRef(event) {
```

Comment:
This should be called 
```suggestion
  onMouseMove(event) {
```